### PR TITLE
rfc-net-20210607-1: Initial Commit of Gem Crash Reporting RFC

### PR DESCRIPTION
**Pull request by lmbr-pip**
Initial RFC for Game Crash Reporting Gem

### CODE REVIEW

File: sigs/sig-network/rfc-net-20210607-1.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    virtual bool UpdateHttpTransport(std::unique_ptr<crashpad::HTTPTransport>& httpTransport, const std::string& baseURL);
+     
+    // Provide hook for argument parsing.
+    virtual void ParseArguments(int& argc, char** argv);
+    // Provide hook for user stream sources.
+    virtual crashpad::UserStreamDataSources* GetUserStreamSources();
+}
+```
+
+####Crash Handler
```
Comment (jeremyong) [inline sigs/sig-network/rfc-net-20210607-1.md:99]: FYI The missing space before the `C` causes a formatting issue here

**Comment by jeremyong** [comment]
Thanks for writing this!

- For the uploader interface, it may be useful to request a pointer to a range of memory that will be supplied in the minidump. On some platforms (consoles), the available size will vary from platform to platform. This enables an upfront allocation or reservation in the data segment so the allocation won't happen during an actual crash scenario
- It's often useful to get a hash of the callstack (+ program counter and link registers) for the purposes of crash deduplication. The hash isn't difficult to calculate but the only minor tricky part is to subtract the base address of the module due to ASLR (address space layout randomization)
-

**Comment by lmbr-pip** [comment]
Am unable to update this RFC at the moment as my fork is no longer attached to https://github.com/o3de/rfcs. Am looking at trying to restore the fork but I may just make a new fork/PR for this that incorporates feedback from @jeremyong.

**Comment by lmbr-pip** [comment]
Closing PR as it needs to either move to new RFC or find new owner. No plans to deliver this work at this point in time.

**Pull request closed without merging**